account_default_draft_move/account_view.xml

```diff
@@ -8,7 +8,12 @@
       <field name="arch" type="xml">
         <data>
           <button name="button_validate" position="replace"/>
-          <button name="button_cancel" position="replace"/>
+          <button name="button_cancel" position="after">
+            <field name="update_posted" invisible="1"/>
+          </button>
+          <button name="button_cancel" position="attributes">
+            <attribute name="attrs">{'invisible': ['|', ('update_posted', '=', False)]}</attribute>
```

Comment:
In the original view (https://github.com/odoo/odoo/blob/7.0/addons/account/account_view.xml#L1268)
there is the following line :
`<button name="button_cancel" states="posted" string="Cancel Entry" type="object" groups="account.group_account_invoice"/>`
This lines is computed with following modifier in the view :
`<button name="button_cancel" states="posted" string="Cancel Entry" type="object" modifiers="{'invisible': [['state', 'not in', ['posted']]]}"/>`

The attrs I add here is prepend to the existing invisible modifier :
`<button name="button_cancel" states="posted" string="Cancel Entry" type="object" attrs="{'invisible': ['|', ('update_posted', '=', False)]}" modifiers="{'invisible': ['|', ['update_posted', '=', false], ['state', 'not in', ['posted']]]}"/>`


What I need  is to set the button invisible in the following case :

move state | update_posted | cancel button invisible
-------------- | ------------------ | ---------------------------
not posted | true | Yes
not posted | false | Yes
posted | true | No
posted | false | Yes

It result in the following expression :
(not posted and update_posted) or (not posted and not update_posted) or (posted and not update_posted)
(not posted and update_posted) or (not posted and not update_posted) => not posted
(not posted and update_posted) or (not posted and not update_posted) or (posted and not update_posted)
=> not posted or (posted and not update_posted)
=> (not posted or posted) and (not posted or not update_posted)
=> **not posted or not update_posted**

So I need "or".